protos/datamodel/common.proto

```diff
@@ -0,0 +1,136 @@
+syntax = "proto3";
+
+import "google/protobuf/timestamp.proto";
+
+package datamodel.v2;
+option go_package = "github.com/opencost/opencost/core/pkg/model/pb;pb";
+
+// TimeWindow represents a time range with resolution for cost data aggregation
+// End time is derived from start + resolution (no redundant end field)
+message TimeWindow {
+  // Start of the time window
+  google.protobuf.Timestamp start = 1;
+
+  // Resolution/granularity of data in this window ("1h", "1d", "10m", etc.)
+  // End time = start + parse_duration(resolution)
+  string resolution = 2;
+}
+
+// ResourceLifetime represents the actual lifecycle of a Kubernetes resource
+// Separate from data collection windows - tracks when resource was created/deleted
+message ResourceLifetime {
+  // When the resource was created
+  google.protobuf.Timestamp createdAt = 1;
+
+  // When the resource was deleted (optional - empty for running resources)
+  google.protobuf.Timestamp deletedAt = 2;
+}
+
+// Provider represents the cloud provider or infrastructure type
+enum Provider {
+  PROVIDER_UNSPECIFIED = 0;
+  PROVIDER_AWS = 1;
+  PROVIDER_GCP = 2;
+  PROVIDER_AZURE = 3;
+  PROVIDER_ON_PREMISES = 4;
+  PROVIDER_ALIBABA = 5;
+  PROVIDER_DIGITALOCEAN = 6;
+  PROVIDER_ORACLE = 7;
+}
+
+// Data source types for tracking how data was collected
+enum DataSourceType {
+  DATA_SOURCE_UNSPECIFIED = 0;
+  DATA_SOURCE_PROMETHEUS = 1;
+  DATA_SOURCE_KUBERNETES_API = 2; // "Promless" source
+  DATA_SOURCE_DCGM = 3; // For GPU devices
+  DATA_SOURCE_CLOUD_BILLING = 4; // For external costs
+  DATA_SOURCE_KSM = 5; // Kubernetes State Metrics
```

Comment:
Where would we use `DATA_SOURCE_KSM` and `DATA_SOURCE_DCGM` as opposed to `PROMETHEUS` or `OPENCOST_COLLECTOR`?